### ISSUE TITLE
Dependency issues upgrading 5.19.0 to 6.0.0

### ISSUE THREAD

**Issue by bdbvb** [telerik/kendo-react#1725]
## I'm submitting a...
Bug report 

## Current behavior
Upgrading from 5.19.0 to 6.0.0

![image](https://github.com/telerik/kendo-react/assets/11095196/16a4c56d-a3f0-4237-9fcd-56f1b78d1272)

npm install gives an error:

```
cmd.exe /c npm install --color=always
npm ERR! code ERESOLVE
ERESOLVE unable to resolve dependency tree
npm ERR! npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!npm ERR!
npm ERR! 
 While resolving: glacier@1.0.0
 Found: @progress/kendo-react-animation@6.0.0
 node_modules/@progress/kendo-react-animation
   @progress/kendo-react-animation@"6.0.0" from the root project
 
 Could not resolve dependency:
 peer @progress/kendo-react-animation@"^5.0.0" from @progress/kendo-react-data-tools@6.0.0
 node_modules/@progress/kendo-react-data-tools
   @progress/kendo-react-data-tools@"6.0.0" from the root project
 
 Fix the upstream dependency conflict, or retry
 this command with --force or --legacy-peer-deps
 to accept an incorrect (and potentially broken) dependency resolution.
```
I can use --legacy-peer-deps, but I'm not sure what that's doing to all my other packages, and it seems to also be required on update as well. This started happening with 5.19.0. 

## Expected behavior
Install cleanly without having to use --legacy-peer-deps for installs and updates so I can have clean and predictable package management.

## Minimal reproduction of the problem with instructions
N/A

## What is the motivation or use case for changing the behavior?
Ability to upgrade properly 

## Environment

Package versions:
list of relevant packages....
```
        "@progress/kendo-data-query": "1.7.0",
        "@progress/kendo-drawing": "1.18.0",
        "@progress/kendo-licensing": "1.3.3",
        "@progress/kendo-react-animation": "6.0.0",
        "@progress/kendo-react-buttons": "6.0.0",
        "@progress/kendo-react-data-tools": "6.0.0",
        "@progress/kendo-react-dateinputs": "6.0.0",
        "@progress/kendo-react-dialogs": "6.0.0",
        "@progress/kendo-react-dropdowns": "6.0.0",
        "@progress/kendo-react-form": "6.0.1",
        "@progress/kendo-react-grid": "6.0.0",
        "@progress/kendo-react-inputs": "6.0.0",
        "@progress/kendo-react-intl": "6.0.0",
        "@progress/kendo-react-layout": "6.0.0",
        "@progress/kendo-react-listview": "6.0.0",
        "@progress/kendo-react-notification": "6.0.0",
        "@progress/kendo-react-popup": "6.0.0",
        "@progress/kendo-react-progressbars": "6.0.0",
        "@progress/kendo-react-tooltip": "6.0.0",
        "@progress/kendo-react-treeview": "6.0.0",
        "@progress/kendo-react-upload": "6.0.0",
        "@progress/kendo-svg-icons": "2.0.0",
        "@progress/kendo-theme-default": "7.0.1",
        "@types/node": "*",
        "@types/react": "18.2.28",
        "@types/react-dom": "18.2.13",
        "react": "18.2.0",
        "react-dom": "18.2.0"
```

System:
- TypeScript version: 5.2.2
- Node version: 
  npm: '10.2.0',
  node: '18.18.0',
- Platform:  Windows

**Comment by nstoychev** [comment]
@bdbvb - Should be fixed in 6.0.2.

**Comment by bdbvb** [comment]
> @bdbvb - Should be fixed in 6.0.2.

Unfortunately not:

<details>
<code>
0 verbose cli C:\Program Files\nodejs\node.exe C:\Users\<redacted>\AppData\Roaming\npm\node_modules\npm\bin\npm-cli.js
1 info using npm@10.2.0
2 info using node@v18.18.0
3 timing npm:load:whichnode Completed in 2ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:C:\Users\<redacted>\AppData\Roaming\npm\node_modules\npm\npmrc Completed in 3ms
6 timing config:load:builtin Completed in 3ms
7 timing config:load:cli Completed in 1ms
8 timing config:load:env Completed in 1ms
9 timing config:load:file:C:\Projects\<redacted>\<redacted>\.npmrc Completed in 1ms
10 timing config:load:project Completed in 1ms
11 timing config:load:file:C:\Users\<redacted>\.npmrc Completed in 1ms
12 timing config:load:user Completed in 1ms
13 timing config:load:file:C:\Users\<redacted>\AppData\Roaming\npm\etc\npmrc Completed in 1ms
14 timing config:load:global Completed in 1ms
15 timing config:load:setEnvs Completed in 1ms
16 timing config:load Completed in 10ms
17 timing npm:load:configload Completed in 10ms
18 timing config:load:flatten Completed in 2ms
19 timing npm:load:mkdirpcache Completed in 0ms
20 timing npm:load:mkdirplogs Completed in 0ms
21 verbose title npm install
22 verbose argv "install" "--color" "always"
23 timing npm:load:setTitle Completed in 1ms
24 timing npm:load:display Completed in 0ms
25 verbose logfile logs-max:10 dir:C:\Users\<redacted>\AppData\Local\npm-cache\_logs\2023-10-12T13_12_51_633Z-
26 verbose logfile C:\Users\<redacted>\AppData\Local\npm-cache\_logs\2023-10-12T13_12_51_633Z-debug-0.log
27 timing npm:load:logFile Completed in 8ms
28 timing npm:load:timers Completed in 0ms
29 timing npm:load:configScope Completed in 0ms
30 timing npm:load Completed in 48ms
31 timing arborist:ctor Completed in 0ms
32 silly logfile start cleaning logs, removing 2 files
33 silly logfile done cleaning log files
34 timing arborist:ctor Completed in 0ms
35 timing idealTree:init Completed in 388ms
36 timing idealTree:userRequests Completed in 0ms
37 silly idealTree buildDeps
38 silly fetch manifest @progress/kendo-react-animation@6.0.2
39 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-animation 16ms (cache hit)
40 silly fetch manifest @progress/kendo-licensing@1.3.3
41 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-licensing 7ms (cache hit)
42 silly fetch manifest react@18.2.0
43 http fetch GET 200 https://registry.npmjs.org/react 10ms (cache hit)
44 silly fetch manifest react-dom@18.2.0
45 http fetch GET 200 https://registry.npmjs.org/react-dom 10ms (cache hit)
46 silly fetch manifest @progress/kendo-react-buttons@6.0.2
47 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-buttons 12ms (cache hit)
48 silly fetch manifest @progress/kendo-svg-icons@2.0.0
49 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-svg-icons 8ms (cache hit)
50 silly fetch manifest @progress/kendo-react-data-tools@6.0.2
51 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-data-tools 9ms (cache hit)
52 silly fetch manifest @progress/kendo-data-query@1.7.0
53 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-data-query 7ms (cache hit)
54 silly fetch manifest @progress/kendo-drawing@1.18.0
55 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-drawing 8ms (cache hit)
56 silly fetch manifest @progress/kendo-react-dateinputs@6.0.2
57 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-dateinputs 9ms (cache hit)
58 silly fetch manifest @progress/kendo-react-intl@6.0.2
59 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-intl 9ms (cache hit)
60 silly fetch manifest @progress/kendo-react-dropdowns@6.0.2
61 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-dropdowns 9ms (cache hit)
62 silly fetch manifest @progress/kendo-react-progressbars@6.0.2
63 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-progressbars 10ms (cache hit)
64 silly fetch manifest @progress/kendo-react-treeview@6.0.2
65 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-treeview 8ms (cache hit)
66 silly fetch manifest @progress/kendo-react-inputs@6.0.2
67 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-inputs 12ms (cache hit)
68 silly fetch manifest @progress/kendo-react-form@^6.0.0
69 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-form 2834ms (cache revalidated)
70 silly fetch manifest @progress/kendo-react-popup@6.0.2
71 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-popup 10ms (cache hit)
72 silly fetch manifest @progress/kendo-react-dialogs@6.0.2
73 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-dialogs 10ms (cache hit)
74 silly fetch manifest @progress/kendo-react-grid@6.0.2
75 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-grid 12ms (cache hit)
76 silly fetch manifest @progress/kendo-react-layout@6.0.2
77 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-layout 14ms (cache hit)
78 silly fetch manifest @progress/kendo-react-listview@6.0.2
79 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-listview 11ms (cache hit)
80 silly fetch manifest @progress/kendo-react-notification@6.0.2
81 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-notification 12ms (cache hit)
82 silly fetch manifest @progress/kendo-react-tooltip@6.0.2
83 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-tooltip 12ms (cache hit)
84 silly fetch manifest @progress/kendo-react-upload@6.0.2
85 http fetch GET 200 https://registry.npmjs.org/@progress%2fkendo-react-upload 10ms (cache hit)
86 silly fetch manifest sass@1.69.3
87 http fetch GET 200 https://registry.npmjs.org/sass 11ms (cache hit)
88 silly placeDep ROOT @progress/kendo-react-animation@6.0.2 REPLACE for: <redacted>@1.0.0 want: 6.0.2
89 silly fetch manifest @progress/kendo-react-common@6.0.2
90 silly placeDep ROOT @progress/kendo-react-buttons@6.0.2 REPLACE for: <redacted>@1.0.0 want: 6.0.2
91 timing idealTree Completed in 3638ms
92 timing command:install Completed in 3642ms
93 verbose stack Error: could not resolve
93 verbose stack     at PlaceDep.failPeerConflict (C:\Users\<redacted>\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\place-dep.js:503:25)
93 verbose stack     at new PlaceDep (C:\Users\<redacted>\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\place-dep.js:157:21)
93 verbose stack     at #buildDepStep (C:\Users\<redacted>\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:933:18)
93 verbose stack     at async Arborist.buildIdealTree (C:\Users\<redacted>\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:196:7)
93 verbose stack     at async Promise.all (index 1)
93 verbose stack     at async Arborist.reify (C:\Users\<redacted>\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\reify.js:159:5)
93 verbose stack     at async Install.exec (C:\Users\<redacted>\AppData\Roaming\npm\node_modules\npm\lib\commands\install.js:152:5)
93 verbose stack     at async module.exports (C:\Users\<redacted>\AppData\Roaming\npm\node_modules\npm\lib\cli-entry.js:61:5)
94 verbose cwd C:\Projects\<redacted>\<redacted>
95 verbose Windows_NT 10.0.22621
96 verbose node v18.18.0
97 verbose npm  v10.2.0
98 error code ERESOLVE
99 error ERESOLVE could not resolve
100 error
101 error While resolving: [1m<redacted>[22m@[1m1.0.0[22m
101 error Found: [1m@progress/kendo-react-data-tools[22m@[1m5.19.0[22m[2m[22m
101 error [2mnode_modules/@progress/kendo-react-data-tools[22m
101 error   [1m@progress/kendo-react-data-tools[22m@"[1m6.0.2[22m" from the root project
101 error   [35mpeer[39m [1m@progress/kendo-react-data-tools[22m@"[1m^5.0.0[22m" from [1m@progress/kendo-react-grid[22m@[1m5.19.0[22m[2m[22m
101 error   [2mnode_modules/@progress/kendo-react-grid[22m
101 error     [1m@progress/kendo-react-grid[22m@"[1m6.0.2[22m" from the root project
101 error
101 error Could not resolve dependency:
101 error [1m@progress/kendo-react-data-tools[22m@"[1m6.0.2[22m" from the root project
101 error
101 error Conflicting peer dependency: [1m@progress/kendo-react-dateinputs[22m@[1m6.0.2[22m[2m[22m
101 error [2mnode_modules/@progress/kendo-react-dateinputs[22m
101 error   [35mpeer[39m [1m@progress/kendo-react-dateinputs[22m@"[1m^6.0.0[22m" from [1m@progress/kendo-react-data-tools[22m@[1m6.0.2[22m[2m[22m
101 error   [2mnode_modules/@progress/kendo-react-data-tools[22m
101 error     [1m@progress/kendo-react-data-tools[22m@"[1m6.0.2[22m" from the root project
101 error
101 error Fix the upstream dependency conflict, or retry
101 error this command with --force or --legacy-peer-deps
101 error to accept an incorrect (and potentially broken) dependency resolution.
102 error
102 error
102 error For a full report see:
102 error C:\Users\<redacted>\AppData\Local\npm-cache\_logs\2023-10-12T13_12_51_633Z-eresolve-report.txt
103 verbose exit 1
104 timing npm Completed in 4164ms
105 verbose unfinished npm timer reify 1697116372149
106 verbose unfinished npm timer reify:loadTrees 1697116372151
107 verbose unfinished npm timer idealTree:buildDeps 1697116372540
108 verbose unfinished npm timer idealTree:#root 1697116372540
109 verbose code 1
110 error A complete log of this run can be found in: C:\Users\<redacted>\AppData\Local\npm-cache\_logs\2023-10-12T13_12_51_633Z-debug-0.log
</code>
</details>

**Comment by bdbvb** [comment]
This was probably just some corruption in node_modules, resolved by deleting package-lock.json and node_modules and re-installing. Closing.
